### PR TITLE
Added simple logger and updated license headers

### PR DESCRIPTION
**Pull request by CraftedMods**
For the future, we can use JUnit tests (with TravisCI), which is useful. This PR is part of a sequence of PRs, which will introduce the core framework of Lembas.

### CODE REVIEW

File: src/main/java/lembas/core/LembasCore.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (C) 2019 Lembas Modding Team
+ * Copyright (C) 2020 Lembas Modding Team
```
Comment (mahtaran) [inline src/main/java/lembas/core/LembasCore.java]: We should use a copyright range (id est `2019-2020`)

File: src/main/java/lembas/core/api/common/SimpleLogger.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package lembas.core.api.common;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * A simple wrapper for a logger using String.format for formatting the log
```
Comment (mahtaran) [inline src/main/java/lembas/core/api/common/SimpleLogger.java]: Using `String.format` is not a necessity, and should thus not be stated as such here.

File: src/main/java/lembas/core/api/common/SimpleLogger.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * messages. The [...]Throwable functions are there to avoid bugs related to the
```
Comment (mahtaran) [inline src/main/java/lembas/core/api/common/SimpleLogger.java]: How does that help?

File: src/test/java/lembas/core/api/common/SimpleLoggerTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ ******************************************************************************/
+package lembas.core.api.common;
+
+import org.junit.Test;
+
+public class SimpleLoggerTest
```
Comment (mahtaran) [inline src/test/java/lembas/core/api/common/SimpleLoggerTest.java:23]: This should probably also contain tests for when nothing is wrong, and a test when you try to create a logger with a null name while a provider is set.

File: src/main/java/lembas/core/api/common/SimpleLogger.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ * varargs.
+ *
+ */
+public abstract class SimpleLogger
```
Comment (mahtaran) [inline src/main/java/lembas/core/api/common/SimpleLogger.java:28]: This could be transformed into an interface for a little more flexibility.

**Review by mahtaran — CHANGES_REQUESTED** [review]

File: src/main/java/lembas/core/LembasCore.java (hunk 1 above)
Comment (CraftedMods) [reply src/main/java/lembas/core/LembasCore.java]: 2019 - currentYear ? I'll adjust that...

File: src/main/java/lembas/core/api/common/SimpleLogger.java (hunk 2 above)
Comment (CraftedMods) [reply src/main/java/lembas/core/api/common/SimpleLogger.java]: What I mean there is that the logger uses `String.format` - this is an requirement for the implementation. The reason why it's here is that the argument syntax is interpreted that way (the formatting specifiers for `String.format` are used) - of course I could make that a provider specific thing, but this bypasses the idea of a really simple logger class and would make the logger creation more complicated.

File: src/main/java/lembas/core/api/common/SimpleLogger.java (hunk 3 above)
Comment (CraftedMods) [reply src/main/java/lembas/core/api/common/SimpleLogger.java]: The problem with the normal logger interfaces (Log4j etc.) is, that they only have varargs, and for example the first one, if it's an exception, will be threated as such one. Now if I, by accident, swap two arguments, the exception will be threated as a normal parameter and may, depending on the message formatting, be included in the message (with a completely wrong meaning then, because it's at the place where the other argument should be), or just swallowed completely. This approach gives more safety, because the exception argument is explicitely needed and not "swallowed" with the varargs, so if the programmer intends that the exception instances is logged, this ensures that it really is.

File: src/test/java/lembas/core/api/common/SimpleLoggerTest.java (hunk 4 above)
Comment (CraftedMods) [reply src/test/java/lembas/core/api/common/SimpleLoggerTest.java:23]: Yeah, if we want to do the full testing, we've to account for a lot of test-cases, which indeed are exploding with more parameters. I just wanted to ensure the most important things, therefore the tests. I'll add the two conditions because they make sence there, but generally, I don't think that we should spend too much effort on the tests, yes, usually that's a good thing to put effort into, but I don't want to make the whole thing test-driven, but only use the tests for a subset of the use-cases.

File: src/main/java/lembas/core/api/common/SimpleLogger.java (hunk 5 above)
Comment (CraftedMods) [reply src/main/java/lembas/core/api/common/SimpleLogger.java:28]: In the PvP Mode Mod it actually is an interface, but then I have to create separate factory classes, because interfaces don't support private members. Therefore I made the class abstract (which btw. shouldn't be a huge downside because the only implementation is probably still the one in our internal package...).

File: src/main/java/lembas/core/api/common/SimpleLogger.java (hunk 2 above)
Comment (mahtaran) [reply src/main/java/lembas/core/api/common/SimpleLogger.java]: In that case I suggest
```suggestion
 * A simple wrapper for a logger that should use String.format for formatting the log
```

File: src/main/java/lembas/core/LembasCore.java (hunk 1 above)
Comment (CraftedMods) [reply src/main/java/lembas/core/LembasCore.java]: Fixed with f7985b49c3aa999b2ce2179deb426572cc1c49d9

File: src/test/java/lembas/core/api/common/SimpleLoggerTest.java (hunk 4 above)
Comment (CraftedMods) [reply src/test/java/lembas/core/api/common/SimpleLoggerTest.java:23]: Fixed with b4c118bc5adbfca296b037f41550b3c982f90834

**Comment by CraftedMods** [comment]
I adressed the things you did mention. Approve the PR (or suggest more changes) when you have time, I'll merge it then.

**Review by mahtaran — APPROVED** [review]
